### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by stuartpb** [opws/opws-dataset#100]
I've been using the "v0.1.0" and "future" milestones for a while now - I think I should probably take inventory on what each of those will entail.

**Comment by stuartpb** [comment]
ALSO, there should really be a document describing how issues / labels / profiling / projects / milestones / the wiki / development flow and interact. Like, I think that kind of supersedes this issue - and, in fact, I think that's a crucial enough part of the overall system operation that I'm putting that as a blocker of 0.1.

**Comment by stuartpb** [comment]
This definitely touches #137

**Comment by stuartpb** [comment]
Also called for: a high-level history of the project linked into that, so older work done before newer best practices can be more easily understood

**Comment by stuartpb** [comment]
What makes sense here as the jumping-off point to all of this is a wiki page linked from the README

**Comment by stuartpb** [comment]
Said wiki page has been written: https://github.com/opws/domainprofiles/wiki/History

**Comment by stuartpb** [comment]
Yeah, a high-level roadmap - or, like, a high-level roadmap that's followed by increasingly lower-level roadmaps - would be nice. I totally forgot what I'd settled on for, like, #42 and #43.

**Comment by stuartpb** [comment]
Like, what I really want is a planfile.

**Comment by stuartpb** [comment]
Note that #147 will dictate what any such documented procedure would look like, significantly.

**Comment by stuartpb** [comment]
Just noted the thing about how extra information is getting stashed in pull request comments (though I *am* storing these facts in a Google Keep window *while I write them*).

**Comment by stuartpb** [comment]
Between this and #147, I'm considering putting a due date on the "policy" milestone to run in parallel with "v0.1.0", meaning that the "policy" milestone must be completed by shelling out to an outside project at that time (ie. not just a bunch of pull quotes here and there scattered across a hundred different issues).

**Comment by stuartpb** [comment]
Another piece of advice that should go in the style guide: many small, simple, and specific fields are better than few complex, vast ones - I've found, over time, that attempts to encapsulate several details into one pattern tends to fall apart in favor of maybe-redundant but orthogonal tests that can be reasoned about easily one by one. They have less baggage when new details come to a profile.

**Comment by stuartpb** [comment]
Regarding how issues should be filed / handled, #22 has just raised an interesting point, regarding the curious interplay between:

- Issues that point out that a problem exists
- Issues that propose a way of solving the problem
- Pull requests that solve issues
- Issues revisiting *older* issues that may have been solved inadequately
- These things getting mixed up in each other
- Labeling things as "accepted" or "declined"

There would stand to be some guidelines (in a "guidelines" repo, perhaps? #147) around this.

**Comment by stuartpb** [comment]
There's also the way that "new fields" has been used as the label for things that aren't represented in the spec right now, even when those things got solved as an *emergent property of a refactor* rather than the addition of any new "field". So maybe those notions should get pulled out of each other, with only *proposals for new fields* getting the "new fields" label, and some new and more general label for "more structured information"?

This is also a salient point for when these remaining schema-centric issues get ported to the schemata issue tracker per #147.

**Comment by stuartpb** [comment]
I'm just going to go ahead and close this in light of the new https://github.com/opws/opws-guidelines repo, and maybe I'll open a new issue in *that* repo that calls back out to this one.
